sdrbase/webapi/webapirequestmapper.cpp

```diff
@@ -4480,116 +4492,139 @@ bool WebAPIRequestMapper::getChannelSettings(
+            channelSettings->getDatvModSettings()->init();
             channelSettings->getDatvModSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "DABDemodSettings")
         {
             channelSettings->setDabDemodSettings(new SWGSDRangel::SWGDABDemodSettings());
+            channelSettings->getDabDemodSettings()->init();
             channelSettings->getDabDemodSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "DOA2Settings")
         {
             channelSettings->setDoa2Settings(new SWGSDRangel::SWGDOA2Settings());
+            channelSettings->getDoa2Settings()->init();
             channelSettings->getDoa2Settings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "DSCDemodSettings")
         {
             channelSettings->setDscDemodSettings(new SWGSDRangel::SWGDSCDemodSettings());
+            channelSettings->getDscDemodSettings()->init();
             channelSettings->getDscDemodSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "DSDDemodSettings")
         {
             channelSettings->setDsdDemodSettings(new SWGSDRangel::SWGDSDDemodSettings());
+            channelSettings->getDsdDemodSettings()->init();
             channelSettings->getDsdDemodSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "FileSinkSettings")
         {
             channelSettings->setFileSinkSettings(new SWGSDRangel::SWGFileSinkSettings());
+            channelSettings->getFileSinkSettings()->init();
             channelSettings->getFileSinkSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "FileSourceSettings")
         {
             channelSettings->setFileSourceSettings(new SWGSDRangel::SWGFileSourceSettings());
+            channelSettings->getFileSourceSettings()->init();
             channelSettings->getFileSourceSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "FreeDVDemodSettings")
         {
             channelSettings->setFreeDvDemodSettings(new SWGSDRangel::SWGFreeDVDemodSettings());
+            channelSettings->getFreeDvDemodSettings()->init();
             channelSettings->getFreeDvDemodSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "FreeDVModSettings")
         {
             channelSettings->setFreeDvModSettings(new SWGSDRangel::SWGFreeDVModSettings());
+            channelSettings->getFreeDvModSettings()->init();
             channelSettings->getFreeDvModSettings()->fromJsonObject(settingsJsonObject);
         }
         else if (channelSettingsKey == "FreqScannerSettings")
         {
             channelSettings->setFreqScannerSettings(new SWGSDRangel::SWGFreqScannerSettings());
+	        channelSettings->getFreqScannerSettings()->init();
```

Comment:
Fix the problem itself. 
Rest of changes are proactively doing the same for the rest of types, to avoid somebody getting into the same trouble